### PR TITLE
Add contributing blog

### PR DESCRIPTION
**Pull request by jimhester**
Things which still probably need to be done

- [x] - I have not added any type of image to the post (suggestions welcome)
- [x] - There should a link to the original post somewhere? *This post originally appeared on http://www.jimhester.com/2017/08/08/contributing/*
- [x] - Any suggestions on re-organization / editing would be good, it could probably be a little more concise...

### CODE REVIEW

**Comment by hadley** [comment]
A few image ideas:

* https://unsplash.com/photos/MPU7_mQB_AY
* https://unsplash.com/photos/qFxS5FkUSAQ
* https://unsplash.com/photos/y83Je1OC6Wc
* https://unsplash.com/photos/UYDoE_AyJQs

I think you should do the credit in a blockquote at the top of the post.

I'll read through and give some more comments on the writing

File: content/articles/2017-08-08-contributing.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+---
+
+Contributing code to open source projects can be intimidating. These projects
+are often widely used and have well known maintainers. Contributing code and
+having it accepted seems an almost insurmountable task.
+
+However if you follow a few simple strategies you can have your code accepted
+into even the most popular projects in the [tidyverse](https://tidyverse.org).
+
+### Don't Contribute Code at all ###
```
Comment (hadley) [inline content/articles/2017-08-08-contributing.md]: Can you please use sentence case?

I think you could probably drop "at all" too

File: content/articles/2017-08-08-contributing.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+accept, so it is a great place to start getting used to the contribution
+process.
+
+### Read CONTRIBUTING.md ###
+
+Often projects will have a `CONTRIBUTING.md` document that has instructions for
+contributing to the project. These are guidelines the maintainers would like
+contributors to adhere to and exist to make the process flow more smoothly. As a
+contributor you should try to make accepting your code as easy as you
+can, this greatly increases the chance your contribution will be accepted.
```
Comment (hadley) [inline content/articles/2017-08-08-contributing.md]: ... These are not currently widespread in the tidyverse, but it's something we'll be working on in the future!

File: content/articles/2017-08-08-contributing.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+Next you should read a few previously merged pull requests for additional
+context. If a project does not have a `CONTRIBUTING.md` (or similar)
+document this is your best source of information on expected practices for
+contributions.
+
+Things you should look for include how are the commit messages formatted? Are
+any additional files changed apart from the code changes (such as NEWS
+updates)? Do the contributions all include additional test cases? Do internal
+only changes need documentation?
```
Comment (hadley) [inline content/articles/2017-08-08-contributing.md]: Maybe note a few common tidyverse convetions? (i.e. add a bullet to `NEWS.md`)

Remind to use `Closes #xyz` to automatically close issue when PR merged?

File: content/articles/2017-08-08-contributing.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+Read the reviewer comments in the pull request
+to get an idea of what in particular reviewers are looking for. Do they require
+certain code style, variable names or code organization? Are there common
```
Comment (hadley) [inline content/articles/2017-08-08-contributing.md]: Link code style to style.tidyverse.org

**Comment by hadley** [comment]
Hmmm, this messes up the articles page: https://deploy-preview-17--tidyverse-org.netlify.com/articles/.  I'll take a look at it tomorrow

File: content/articles/2017-08-08-contributing.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
+---
+title: Contributing Code to the Tidyverse
```
Comment (hadley) [inline content/articles/2017-08-08-contributing.md]: Oh and sentence case here

File: content/articles/2017-08-08-contributing.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+author: Jim Hester
+slug: contributing
+date: 2018-08-08
+photo:
+  url: https://unsplash.com/photos/qFxS5FkUSAQ
+  author: Yuriy Rzhemovskiy
+categories: [Other]
+---
```
Comment (jimhester) [inline content/articles/2017-08-08-contributing.md]: Add

`description:  short description about the article`

**Comment by hadley** [comment]
And you'll need to merge/rebase to get the fixes to the articles page.

**Comment by jennybc** [comment]
Just re-read this in light of my own PR to the Help page:

  * "tidyverse" vs "Tidyverse": It feels like we're converging on "tidyverse" but there uppercase version is used here.
  * Suggest you delete or replace the "one night stand" bit in "View contributing as a relationship, not a one night stand". Possible replacement: "View contributing as a relationship, not a (?one-time?) transaction"
  * Do you want to recommend that someone watch a couple of repos? I think you can absorb a lot of conventions just by lurking for a while.

**Comment by jimhester** [comment]
- I used capital T just because my editor treated it as misspelled if it was not a proper noun, happy to change if that is the style we want to use.
- `View contributing as a relationship, not a transaction` sounds good, thank you for the suggestion!
- I don't really ever watch repos I am not personally maintaining, but I can see how that could be a useful approach...

**Comment by jennybc** [comment]
Yeah, I also started out with capital "T" "Tidyverse" but then noticed what @hadley was doing elsewhere. We need a rule.

Watching is how I absorbed a lot of soft info on how people conducted themselves, before I jumped in. I think it's sounds awful once you are already overwhelmed with GitHub notifications, but if you're just getting started, it can be useful.

File: content/articles/2017-08-08-contributing.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+title: Contributing Code to the tidyverse
```
Comment (hadley) [inline content/articles/2017-08-08-contributing.md]: code

**Comment by hadley** [comment]
I think lowercase is tidier ;)

**Comment by hadley** [comment]
Can you please merge/rebase? It doesn't look like my fixes to the articles page are coming through.

And I guess update the date?

**Comment by jimhester** [comment]
Done

**Comment by hadley** [comment]
And once more? I think I actually fixed it this time

**Comment by hadley** [comment]
Thanks! I'll tweet about it

**Pull request merged**